### PR TITLE
Api gateway

### PR DESCRIPTION
**Pull request by balintzs**
Changes
=======
* Add API Gateway manager
  * Rest API creation with swaggerfile
* Add API Gateway integration to Lambda manager
  * dynamically adding permission to invoke lambda by API Gateway

Testing
======
* Unit testing with `botocore.stubber`
* Tested in production

### CODE REVIEW

File: aws_steps/api_gateway/api_gateway.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            description = name
+
+        print("Initializing new API: {} ...".format(name))
+        return client.create_rest_api(
+            name=name,
+            description=description
+        )
+
+    @classmethod
+    def delete_api_gateway(cls, name):
```
Comment (paulzah) [inline aws_steps/api_gateway/api_gateway.py]: what is the use-case for deleting an API-Gateway?

**Comment by paulzah** [comment]
why use a stubber in tests instead of sticking to moto?

**Comment by balintzs** [comment]
Because it prevents us from having to deal with potential shortcomings of moto. I also plan to add missing tests of lambda using stubber.

File: aws_steps/api_gateway/api_gateway.py (hunk 1 above)
Comment (balintzs) [inline aws_steps/api_gateway/api_gateway.py]: Nothing. I shall remove it.

**Comment by paulzah** [comment]
LGTM

**Pull request merged**